src/theory/quantifiers/sygus/transition_inference.cpp

```diff
@@ -0,0 +1,586 @@
+    {
+      ret = d_com[i].d_conjuncts[0];
+    }
+    else
+    {
+      ret = nm->mkNode(AND, d_com[i].d_conjuncts);
+    }
+    if (i == 0 || i == 1)
+    {
+      // pre-condition and transition are negated
+      ret = TermUtil::simpleNegate(ret);
+    }
+    d_com[i].d_this = ret;
+  }
+}
+void TransitionInference::getNormalizedSubstitution(
+    Node curr,
+    const std::vector<Node>& pvars,
+    std::vector<Node>& vars,
+    std::vector<Node>& subs,
+    std::vector<Node>& disjuncts)
+{
+  for (unsigned j = 0, nchild = curr.getNumChildren(); j < nchild; j++)
+  {
+    if (curr[j].getKind() == BOUND_VARIABLE)
+    {
+      // if the argument is a bound variable, add to the renaming
+      vars.push_back(curr[j]);
+      subs.push_back(pvars[j]);
+    }
+    else
+    {
+      // otherwise, treat as a constraint on the variable
+      // For example, this transforms e.g. a precondition clause
+      // I( 0, 1 ) to x1 != 0 OR x2 != 1 OR I( x1, x2 ).
+      Node eq = curr[j].eqNode(pvars[j]);
+      disjuncts.push_back(eq.negate());
+    }
+  }
+}
+
+bool TransitionInference::processDisjunct(
+    Node n,
+    std::map<bool, Node>& terms,
+    std::vector<Node>& disjuncts,
+    std::map<bool, std::map<Node, bool> >& visited,
+    bool topLevel)
+{
+  if (visited[topLevel].find(n) != visited[topLevel].end())
+  {
+    return true;
+  }
+  visited[topLevel][n] = true;
+  bool childTopLevel = n.getKind() == OR && topLevel;
+  // if another part mentions UF or a free variable, then fail
+  bool lit_pol = n.getKind() != NOT;
+  Node lit = n.getKind() == NOT ? n[0] : n;
+  if (lit.getKind() == APPLY_UF)
+  {
+    Node op = lit.getOperator();
+    if (d_func.isNull())
+    {
+      d_func = op;
+      Trace("cegqi-inv-debug") << "Use " << op << " with args ";
+      NodeManager* nm = NodeManager::currentNM();
+      for (const Node& l : lit)
+      {
+        Node v = nm->mkSkolem("i", l.getType(), "template inference argument");
+        d_vars.push_back(v);
+        Trace("cegqi-inv-debug") << v << " ";
+      }
+      Trace("cegqi-inv-debug") << std::endl;
+    }
+    if (op != d_func)
+    {
+      Trace("cegqi-inv-debug")
+          << "...failed, free function : " << n << std::endl;
+      return false;
+    }
+    else if (topLevel)
+    {
+      if (terms.find(lit_pol) == terms.end())
+      {
+        terms[lit_pol] = lit;
+        return true;
+      }
+      else
+      {
+        Trace("cegqi-inv-debug")
+            << "...failed, repeated inv-app : " << lit << std::endl;
+        return false;
+      }
+    }
+    Trace("cegqi-inv-debug")
+        << "...failed, non-entailed inv-app : " << lit << std::endl;
+    return false;
+  }
+  else if (topLevel && !childTopLevel)
+  {
+    disjuncts.push_back(n);
+  }
+  for (const Node& nc : n)
+  {
+    if (!processDisjunct(nc, terms, disjuncts, visited, childTopLevel))
+    {
+      return false;
+    }
+  }
+  return true;
+}
+
+Node TransitionInference::getComponent(int i) const
```

Comment:
redo this to reflect the new shape of d_com. Probably use an assigned from 0 to 2 so that you can just test if it is in that range to return `null`

Reply:
Function was deleted.